src/core/operators/vid_vec_rep_clip.py

```diff
@@ -0,0 +1,157 @@
+            self.feature_matrix = []
+            self.analyze(fname)
+
+        def get_mean_feature(self):
+            """
+            Returns:
+                torch.Tensor: Mean feature vector
+            """
+            return torch.mean(self.feature_matrix, dim=0)
+
+        def analyze(self, fname):
+            """
+            Analyzes the video file and extracts features.
+
+            Args:
+                fname (str): Path to the video file
+
+            Raises:
+                FileNotFoundError: If the file is not found
+            """
+            # check if file exists
+            if not os.path.exists(fname):
+                raise FileNotFoundError(f"File not found: {fname}")
+
+            # Extract I-frames and features
+            self.frame_images = self.extract_frames(fname)
+            print(len(self.frame_images))
+            self.feature_matrix = self.extract_features(self.frame_images)
+
+        def extract_frames(self, fname):
+            """
+            Extracts I-frames from the video file using `ffmpeg`.
+
+            Args:
+                fname (str): Path to the video file
+
+            Returns:
+                list: List of PIL Images
+            """
+            with tempfile.TemporaryDirectory() as temp_dir:
+                # Command to extract I-frames using ffmpeg's command line tool
+                cmd=f"""
+                ffmpeg -i "{fname}" -vf "select=eq(pict_type\,I)" -vsync vfr "{temp_dir}/frame_%05d.jpg"
+                """
+                with subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE) as process:
+                    process.wait()
+                frames = []
+                for filename in os.listdir(temp_dir):
+                    if filename.endswith((".jpg")):
+                        image_path = os.path.join(temp_dir, filename)
+                        with Image.open(image_path) as img:
+                            frames.append(img.copy())
+                return frames
+
+        def extract_features(self, images):
+            """
+            Extracts features from a list of images using pre-trained CLIP-ViT-B-32.
+
+            Args:
+                images (list): List of PIL Images
+
+            Returns:
+                torch.Tensor: Feature matrix of shape (batch, 512)
+            """
+            inputs = processor(images=images, return_tensors="pt", padding=True, truncation=True)
+            inputs = {k: v.to(self.device) for k, v in inputs.items()}  # move to device
+            with torch.no_grad():
+                features = self.model.get_image_features(**inputs)
+                return features
+
+def run(file):
+    """
+    Runs the operator.
+
+    Args:
+        file (dict): `VideoFactory` file object
+
+    Returns:
+        generator: Yields video and I-frame vector representations
+    """
+    fname = file["path"]
+    vid_analyzer = VideoAnalyzer(fname)
+    return gendata(vid_analyzer)
```

Comment:
can you add a `context manager` here, which delete's the file safely after the processing is done. 

Here is an example of how we are using it in another operator - https://github.com/tattle-made/feluda/blob/2b9275a134677a1338dfbd5f7c9fcd9922c7b447/src/core/operators/media_file_hash.py#L12-L21

Reply:
I am not opening any file in the `run` function. The `ffmpeg` CLI directly takes the file path on disk and exports the I-frames (which we open in the `extract_frames` method

Reply:
right, so that is what I meant, its fine, if you are not opening the file, but something like once you pass it to `VideoAnalyzer` and after its done processing, we safely remove the file. 

I have made a similar implementation in the `resent operator`. - https://github.com/tattle-made/feluda/blob/2b9275a134677a1338dfbd5f7c9fcd9922c7b447/src/core/operators/vid_vec_rep_resnet.py#L230-L260